src/libtiled/tilelayer.h

```diff
@@ -568,6 +571,19 @@ inline void TileLayer::setSize(const QSize &size)
     mHeight = size.height();
 }
 
+inline QRect TileLayer::bounds() const
+{
+    QRect bound = mBounds.boundingRect();
+
+    int x = std::min(0, bound.x());
+    int y = std::min(0, bound.y());
+    int width = std::max(mWidth, bound.width());
+    int height = std::max(mHeight, bound.height());
```

Comment:
Why is this? When a layer is empty, by all means its bounds should be empty as well. That is actually a nice optimization for the code using the `bounds` method.